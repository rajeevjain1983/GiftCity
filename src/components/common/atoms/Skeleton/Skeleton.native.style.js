// 9fbef606107a605d69c0edbcd8029e5d 
import styled from 'styled-components';

const getRowStyle = props => {
  const { rowProps } = props;
  return rowProps;
};
const getColStyle = props => {
  const { colProps } = props;
  return colProps;
};

export const Row = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  ${getRowStyle}
`;
export const Col = styled.View`
  margin-bottom: 5px;
  margin-right: 5px;
  background: #d8d8d8;
  position: relative;
  ${getColStyle}
  ${props => (props.outFitSkeleton ? `margin:12px 12px; ` : '')}
`;
