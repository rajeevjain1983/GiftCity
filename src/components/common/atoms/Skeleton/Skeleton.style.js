// 9fbef606107a605d69c0edbcd8029e5d 
import { css, keyframes } from 'styled-components';

const progress = keyframes`0% {
  left: -500px;
}
100% {
  left: 500px;
}`;

const style = css`
  &.skeleton-row {
    position: relative;
  }
  .skeleton-col {
    height: 150px;
    ${props =>
      props.removeLastMargin
        ? `&:nth-child(${props.col}){
      margin-right:0;
    }`
        : ''}
    background-color: rgb(239, 241, 246);
    overflow: hidden;
    position: relative;
    &:before {
      content: '';
      position: absolute;
      height: 100%;
      width: 100%;
      top: 0;
      left: -500px;
      background-image: linear-gradient(
        90deg,
        rgba(255, 255, 255, 0),
        rgba(255, 255, 255, 0.6),
        rgba(255, 255, 255, 0)
      );
      animation: ${progress} 0.5s ease-in-out infinite;
    }
  }
  .left-carousel {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: 0;
    display: none;
    @media ${props => props.theme.mediaQuery.large} {
      display: block;
    }
  }
  .right-carousel {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    right: 0;
    display: none;
    @media ${props => props.theme.mediaQuery.large} {
      display: block;
    }
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
export default style;
