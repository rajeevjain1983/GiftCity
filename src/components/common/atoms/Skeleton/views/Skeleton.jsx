// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '../../../hoc/withStyles';
import style from '../Skeleton.style';
import { getIconPath } from '../../../../../utils';
import Image from '../../Image';
import Row from '../../Row';
import Col from '../../Col';

const Skeleton = ({ className, col, showArrows, colSize, removeLastMargin, labels }) => {
  const rightCarouselPath = getIconPath('carousel-big-carrot');
  const leftCarouselPath = getIconPath('carousel-big-carrot-left');
  return (
    <Row className={`${className} skeleton-row`}>
      {Array.from({ length: col }).map((cItem, index) => {
        return (
          <Col
            key={index.toString()}
            colSize={colSize}
            className="skeleton-col"
            removeLastMargin={removeLastMargin}
          />
        );
      })}
      {showArrows && (
        <Image
          src={leftCarouselPath}
          alt={labels ? labels.accessibility.ariaPrevious : ''}
          className="left-carousel"
        />
      )}
      {showArrows && (
        <Image
          src={rightCarouselPath}
          alt={labels ? labels.accessibility.ariaNext : ''}
          className="right-carousel"
        />
      )}
    </Row>
  );
};

Skeleton.propTypes = {
  className: PropTypes.string.isRequired,
  rowProps: PropTypes.shape({}),
  colProps: PropTypes.shape({}),
  col: PropTypes.number,
  colSize: PropTypes.shape({}),
  showArrows: PropTypes.bool,
  removeLastMargin: PropTypes.bool,
  labels: PropTypes.shape({}).isRequired,
};

Skeleton.defaultProps = {
  rowProps: {},
  colProps: {},
  col: 1,
  colSize: { large: 12, medium: 8, small: 6 },
  showArrows: false,
  removeLastMargin: false,
};

export default withStyles(Skeleton, style);
export { Skeleton as SkeletonVanilla };
