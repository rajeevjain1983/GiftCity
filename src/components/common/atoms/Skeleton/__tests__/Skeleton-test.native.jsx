// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { SkeletonVanilla as Skeleton } from '../views/Skeleton.native';

describe('Home Skeleton Native Componenet', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Skeleton col={3} row={1} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
