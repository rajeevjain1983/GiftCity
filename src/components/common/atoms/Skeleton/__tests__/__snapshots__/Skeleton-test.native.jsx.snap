// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Skeleton Native Componenet should match snapshot 1`] = `
<View>
  <Styled(View)
    key="0"
    rowProps={Object {}}
  >
    <Styled(View)
      colProps={Object {}}
      height={150}
      key="0"
      otherprops={Object {}}
      width={100}
    />
    <Styled(View)
      colProps={Object {}}
      height={150}
      key="1"
      otherprops={Object {}}
      width={100}
    />
    <Styled(View)
      colProps={Object {}}
      height={150}
      key="2"
      otherprops={Object {}}
      width={100}
    />
  </Styled(View)>
</View>
`;
